go/app/main.go

```diff
@@ -25,18 +40,74 @@ func root(c echo.Context) error {
 	return c.JSON(http.StatusOK, res)
 }
 
+func dataJson() Items{
+	//Open our jsonFile
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		fmt.Println(err)
+	}
+	//Defer the closing of our jsonFile so that we can parse it later on
+	defer jsonFile.Close()
+
+	//Read our opened jsonFile as a byte array.
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+
+	//Inicialize our array of items
+	var beforeItems Items
+
+	//Save data into the array
+	json.Unmarshal(byteValue, &beforeItems)
+
+	return beforeItems
+}
+
 func addItem(c echo.Context) error {
-	// Get form data
+	//Get form data
 	name := c.FormValue("name")
-	c.Logger().Infof("Receive item: %s", name)
+	category := c.FormValue("category")
+	imagePath := c.FormValue("image")
+	//Read the data of the image
+	imageData, _ := ioutil.ReadFile(imagePath)
+
+	//Create new image name with sha256
+    newImageName := fmt.Sprintf("%x%s", sha256.Sum256(imageData), ".jpg")
 
-	message := fmt.Sprintf("item received: %s", name)
+	//Message
+	c.Logger().Infof("We recived a %s from category: %s", name, category)
+	message := fmt.Sprintf("We recived a %s from category: %s", name, category)
 	res := Response{Message: message}
+	
+	//Create new item
+	newItem := Item{}
+	newItem.Name = name
+	newItem.Category = category
+	newItem.Image = newImageName
+
+	items := dataJson()
+
+	//Add new 
+	items.Items = append(items.Items, newItem)
+
+	//Save into a JSON file
+	content, err := json.Marshal(items)
+	if err != nil {
+		fmt.Println(err)
+	}
+	err = ioutil.WriteFile("items.json", content, 0644)
```

Comment:
There is one more advanced challenge here. I don't think you need to consider it for this exercise. But try to give it some thought. Our server normally will accept multiple requests in parallel. This means multiple calls to our function may be reading/writing to the same file. Consider what would happen if two requests to add items happened simultaneously. What would happen to the file in this case?